### PR TITLE
Feat: Notional Transfers API and Frontend Component - 606 638

### PR DESCRIPTION
**Pull request by AlexZorkin**

### CODE REVIEW

File: backend/lcfs/web/api/notional_transfer/services.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+        return NotionalTransfersSchema(
+            pagination=PaginationResponseSchema(
+                total=total_count,
+                page=pagination.page,
+                size=pagination.size,
+                total_pages=math.ceil(total_count / pagination.size),
+            ),
+            notional_transfers=[
+                NotionalTransferSchema.model_validate(notional_transfer) for notional_transfer in notional_transfers
```
Comment (hamed-valiollahi) [inline backend/lcfs/web/api/notional_transfer/services.py:51]: If there are no notional transfers, will this cause any errors? Can `NotionalTransfersSchema` handle an empty list of `notional_transfers`?

File: backend/lcfs/web/api/notional_transfer/validation.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from fastapi import Depends, HTTPException, Request
```
Comment (hamed-valiollahi) [inline backend/lcfs/web/api/notional_transfer/validation.py:1]: Nice to have this validation class! It keeps validation logic organized and centralized. 👍

**Review by hamed-valiollahi — COMMENTED** [review]
Although it's a draft PR, the code is great. Thank you, Alex!

File: frontend/src/views/NotionalTransfers/_schema.jsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  {
+    field: 'legalName',
+    headerName: t('notionalTransfer:notionalTransferColLabels.legalName'),
+    cellEditor: 'agTextCellEditor',
+    cellDataType: 'text',
+    cellStyle: (params) => {
+      if (params.data.modified && (!params.value || params.value === ''))
+        return { borderColor: 'red' }
+    },
+    cellRenderer: (params) =>
```
Comment (prv-proton) [inline frontend/src/views/NotionalTransfers/_schema.jsx:50]: Since this is a free form text input, do we need this renderer?

**Pull request merged**